metadata_service/config.py

```diff
@@ -118,37 +114,30 @@ class LocalConfig(LocalGremlinConfig, Config):
     }
 
 
-# The databuilder expects this to be False currently. We are defaulting to true because the testing expects this
-if bool(distutils.util.strtobool(os.environ.get('IGNORE_NEPTUNE_SHARD', 'False'))):
-    shard_set_explicitly('')
+class GremlinConfig(LocalConfig, LocalGremlinConfig):
+    JANUS_GRAPH_URL = None
+
+
+class NeptuneConfig(LocalGremlinConfig, LocalConfig):
+    import boto3    # isort:skip
 
+    def __init__(self) -> None:
+        from amundsen_gremlin.test_and_development_shard import shard_set_explicitly    # isort:skip
+        # The databuilder expects this to be False currently. We are defaulting to true because the testing expects this
+        if bool(distutils.util.strtobool(os.environ.get('IGNORE_NEPTUNE_SHARD', 'False'))):
```

Comment:
Github might be showing this weird but this will not work inside the init. 

The set_shard will only work when it has not been used yet. 
https://github.com/amundsen-io/amundsengremlin/blob/master/amundsen_gremlin/test_and_development_shard.py#L53

and as pointed out by the issue 
https://github.com/amundsen-io/amundsengremlin/blob/master/amundsen_gremlin/test_and_development_shard.py#L74

makes it used.

Reply:
+1 for this approach, we should never rely on functions to be called implicitly on imports anyways

Reply:
@AndrewCiambrone I think this will work as expected. Would be nice if you can please give this a try. 
On the other hand, I believe you already answered yourself how to fix this properly. these kinds of functions should not be called globally and may cause side effects like these.

Reply:
Hey @verdan I pulled down the branch and tried it locally. It unfortunately did not work. It looks like the `__init__ `is not called upon. 
Digging in deeper it looks like because the config is never initialized inside the from_object function: https://github.com/pallets/flask/blob/master/src/flask/config.py#L161

Reply:
I am not familiar enough with the test_and_development_shard.py to know if we can just remove `_reset_for_testing_only()` from being called implicitly. Maybe @friendtocephalopods  might know? 

But a short term solution could be:
- To set the default of `IGNORE_NEPTUNE_SHARD` to be `True` 
- Or move the Neptune config to its own file.

Reply:
@mgorsk1 @AndrewCiambrone how do you think we should proceed with this? 
For a quick fix, we can do this in following ways:
- First, as Andrew mentioned, to set this to True. 
- Second, `if os.environ.get('PROXY_CLIENT', 'NEO4J') == 'NEPTUNE': then the check, and import shard_set_explicitly('')`

Thoughts?

Reply:
My thoughts are we should remove this line https://github.com/amundsen-io/amundsengremlin/blob/master/amundsen_gremlin/test_and_development_shard.py#L74 , add explicit call in tests (where it's actually requried) and bump gremlin dependency. But I don't know this proxy well enough to asses what would be the impact of removing this line.

Reply:
@mgorsk1 @verdan Until someone someone who is more familiar with the testing framework can vouch that removing line 74 is okay I am proposing: https://github.com/amundsen-io/amundsengremlin/pull/13

Reply:
Sure, that change (#13) should be fine! You don't _need_ to shard anything here by user; it's just convenient if you want to have one neptune instance backing multiple tests at once. Defaulting to some static value works.